packages/eslint-config/.eslintrc.js

```diff
@@ -0,0 +1,66 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/eslint-config
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * Default ESLint configuration for LoopBack
+ *
+ * See https://eslint.org/docs/user-guide/configuring
+ */
+module.exports = {
+  root: true,
+  parser: '@typescript-eslint/parser',
+  plugins: ['eslint-plugin', '@typescript-eslint'],
+  env: {
+    es6: true,
+    node: true,
+  },
+  extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended'],
+  rules: {
+    'comma-dangle': ['error', 'always-multiline'],
+    'no-mixed-operators': 'error',
+    'no-console': 'off',
+    'no-undef': 'off',
+    'no-inner-declarations': 'off',
+    // TypeScript allows method overloading
+    'no-dupe-class-members': 'off',
+    'no-useless-escape': 'off',
+    // TypeScript allows the same name for namespace and function
+    'no-redeclare': 'off',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/no-explicit-any': 'error',
+    '@typescript-eslint/no-non-null-assertion': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-member-accessibility': 'off',
+    '@typescript-eslint/no-var-requires': 'off',
+    '@typescript-eslint/no-use-before-define': 'off',
+    '@typescript-eslint/no-object-literal-type-assertion': 'off',
+    '@typescript-eslint/no-parameter-properties': 'off',
+    '@typescript-eslint/no-angle-bracket-type-assertion': 'off',
+    '@typescript-eslint/prefer-interface': 'off',
+    '@typescript-eslint/no-namespace': 'off',
+
+    // Disable the base `no-unused-vars` to use `typescript`
+    'no-unused-vars': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      // Turn it off for now before we find out how it applies to decorators
+      'off',
+      {
+        vars: 'all',
+        args: 'none',
+        ignoreRestSiblings: false,
+      },
+    ],
+    '@typescript-eslint/ban-types': 'error',
```

Comment:
Let's remove this rule from this initial pull request. Our current tslint config does not enforce `ban-types`.

Reply:
Here is my approach:

1. Use the recommended rules from `eslint` and `typescript-eslint` to expose all violations in our code base
2. Fix violations if they make good sense
3. Disable certain rules if we need to discuss or spend more time to fix

Reply:
> Use the recommended rules from `eslint` and `typescript-eslint` to expose all violations in our code base

I see the merit of this. While I am not very happy that it increases the size of this pull request, I guess I can live with that.

However, I feel it's important to understand which rules enabled by our current configuration will be lost after your change. Quoting from an earlier comment:

> Have you verified that all rules we are currently using in `@loopback/tslint-config` already have their counterpart in `@typescript-eslint` or other eslint plugins?
> 
> If there are any rules that are not available yet, then we should have a discussion whether we are ok to accept that limitation.

The current TSLint config was also based on the setup recommended at the time we created them. What's more important, it contains rules that helper us to detect important bad practices in the past. I don't want to loose them.

What I would like to see:
 - List of rules that are enabled now but won't be applied after your change
 - If there are any GH issues in eslint/tslint projects that are tracking these missing rules, then please reference them in the list.
 - The PR description should be updated with this information.

Reply:
BTW, if `ban-types` is a recommended rule, then why do we have to explicitly enable it here? Isn't it automatically inherited from the recommended configuration?